src/program.js

```diff
@@ -146,6 +146,9 @@ export class Program {
           version: getVersion(absolutePackageDir),
         });
       }
+      if (cmd === 'lint') {
+        argv.runAsBinary = this.shouldExitProgram;
```

Comment:
By explicitly setting `runAsBinary` you are making `program.js` care and understand about how the `lint` function operates. It should neither care nor understand :)

Instead, I think it would be better to pass `shouldExitProgram` as an option to `lint` and, actually, to all commands. That would look like:

````js
await runCommand(argv, {shouldExitProgram});
````

With that new optional parameter the `lint` function could decide what to do.

As part of this change, all command handler functions (every `default` export within [`src/cmd`](https://github.com/mozilla/web-ext/tree/master/src/cmd)) will need to accept a boolean `shouldExitProgram` parameter. They can all just ignore it but the Flow object definition should at least define it as a possible parameter.